src/scripts/mod.rs

```diff
@@ -0,0 +1,335 @@
+                            script.tags.clone().unwrap().into_iter().collect();
+                        if config_hashset.is_subset(&script_hashset) {
+                            scripts_to_run.push(script.to_owned());
+                        }
+                        debug!(
+                            "\nScript tags does not match config tags {:?} {}",
+                            &script_hashset,
+                            script.path.clone().unwrap().display()
+                        );
+                    }
+                }
+            }
+            debug!("\nScript(s) to run {:?}", scripts_to_run);
+            Ok(scripts_to_run)
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct Script {
+    // Path to the script itself.
+    path: Option<PathBuf>,
+
+    // Ip got from scanner.
+    ip: IpAddr,
+
+    // Ports found with portscan.
+    open_ports : Vec<u16>,
+
+    // Port found in ScriptFile, if defined only this will run with the ip.
+    port: Option<String>,
+
+    // Character to join ports in case we want to use a string format of them, for example nmap -p.
+    ports_separator: Option<String>,
+
+    // Tags found in ScriptFile.
+    tags: Option<Vec<String>>,
+
+    // The format how we want the script to run.
+    call_format: Option<String>,
+}
+
+#[derive(Serialize)]
+struct ExecPartsScript {
+    script: String,
+    ip: String,
+    port: String,
+}
+
+#[derive(Serialize)]
+struct ExecParts {
+    ip: String,
+    port: String,
+}
+
+impl Script {
+    pub fn build(
+        path: Option<PathBuf>,
+        ip: IpAddr,
+        open_ports : Vec<u16>,
+        port: Option<String>,
+        ports_separator: Option<String>,
+        tags: Option<Vec<String>>,
+        call_format: Option<String>,
+    ) -> Self {
+        Self {
+            path: path,
+            ip: ip,
+            open_ports : open_ports ,
+            port: port,
+            ports_separator: ports_separator,
+            tags: tags,
+            call_format: call_format,
+        }
+    }
+
+    // Some variables get chnaged before read, and compiler throws warning on warn(unused_assignments)
+    #[allow(unused_assignments)]
+    pub fn run(self) -> Result<String> {
+        debug!("run self {:?}", &self);
+
+        let mut ports_str = String::new();
+        if let Some(ports_separator) = self.ports_separator {
+            let ports_str_vec: Vec<String> =
+                self.open_ports .iter().map(|port| port.to_string()).collect();
+            ports_str = ports_str_vec.join(&ports_separator);
+        } else if let None = self.ports_separator {
+            let ports_str_vec: Vec<String> =
+                self.open_ports .iter().map(|port| port.to_string()).collect();
+            ports_str = ports_str_vec.join(",");
+        } else if let Some(port) = self.port {
+            ports_str = port;
+        } else {
+            return Err(anyhow!("Failed to parse ports_separator in scriptfile."));
+        }
```

Comment:
Do we ever reach those since we are already taking into account the `Some` and `None` cases of `self.ports_separator`?

Reply:
The separator got simplified to `let separator = self.ports_separator.unwrap_or(",".into());` based on your other comment. So now we can get here.